Installation issue: ntpoly: missing C build dependency causes CMake compiler test failure

### Steps to reproduce the issue

```console
$ spack spec ntpoly%gcc@14.2.0 ^openblas@0.3.28
 -   ntpoly@3.1.0~ipo+shared build_system=cmake build_type=Release generator=make platform=linux os=rhel8 target=x86_64_v3 %cxx,fortran=gcc@14.2.0
[e]      ^cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release platform=linux os=rhel8 target=x86_64 
 -       ^compiler-wrapper@1.0 build_system=generic platform=linux os=rhel8 target=x86_64_v3 
[e]      ^gcc@14.2.0+binutils+bootstrap~graphite+libsanitizer~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=rhel8 target=x86_64 
 -       ^gcc-runtime@14.2.0 build_system=generic platform=linux os=rhel8 target=x86_64_v3 
[e]      ^glibc@2.28 build_system=autotools platform=linux os=rhel8 target=x86_64 
[e]      ^gmake@4.2.1~guile build_system=generic platform=linux os=rhel8 target=x86_64 
[e]      ^openblas@0.3.28~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none platform=linux os=rhel8 target=x86_64 
[e]      ^openmpi@5.0.6+atomics~cuda~debug+fortran~gpfs~internal-hwloc~internal-libevent~internal-pmix~ipv6~java+lustre~memchecker~openshmem~rocm+romio+rsh~static~two_level_namespace+vt~wrapper-rpath build_system=autotools fabrics:=ucx romio-filesystem:=none schedulers:=none platform=linux os=rhel8 target=x86_64
$ spack install ntpoly%gcc@14.2.0
...
```


### Error message

<details><summary>Error message</summary>

<pre>
4 errors found in build log:
     12    -- The C compiler identification is unknown
     13    -- The CXX compiler identification is GNU 14.2.0
     14    -- Detecting C compiler ABI info
     15    -- Detecting C compiler ABI info - failed
     16    -- Check for working C compiler: /scratch/project_2016753/sarastel/spack/opt/spack/linux-x86_64_v3/compiler-wrapper-1.0-sfqaortqrxragwtt6d6k5ryxrexrjkgy/libexec/spack/cc
     17    -- Check for working C compiler: /scratch/project_2016753/sarastel/spack/opt/spack/linux-x86_64_v3/compiler-wrapper-1.0-sfqaortqrxragwtt6d6k5ryxrexrjkgy/libexec/spack/cc - broken
  >> 18    CMake Error at /appl/spack/v023/install-tree/gcc-8.5.0/cmake-3.31.9-c3gdj2/share/cmake-3.31/Modules/CMakeTestCCompiler.cmake:67 (message):
     19      The C compiler
     20    
     21        "/scratch/project_2016753/sarastel/spack/opt/spack/linux-x86_64_v3/compiler-wrapper-1.0-sfqaortqrxragwtt6d6k5ryxrexrjkgy/libexec/spack/cc"
     22    
     23      is not able to compile a simple test program.
     24    

     ...

     28    
     29        Run Build Command(s): /appl/spack/v023/install-tree/gcc-8.5.0/cmake-3.31.9-c3gdj2/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile cmTC_6390d/fast
     30        /usr/bin/gmake  -f CMakeFiles/cmTC_6390d.dir/build.make CMakeFiles/cmTC_6390d.dir/build
     31        gmake[1]: Entering directory '/local_scratch/sarastel/spack-stage/spack-stage-ntpoly-3.1.0-asg2mljsf3vzgz4wisusxbykg5veaku2/spack-build-asg2mlj/CMakeFiles/CMakeScratch/TryCompile-xTeIn1'
     32        Building C object CMakeFiles/cmTC_6390d.dir/testCCompiler.c.o
     33        /scratch/project_2016753/sarastel/spack/opt/spack/linux-x86_64_v3/compiler-wrapper-1.0-sfqaortqrxragwtt6d6k5ryxrexrjkgy/libexec/spack/cc    -o CMakeFiles/cmTC_6390d.dir/testCCompiler.c.o -
           c /local_scratch/sarastel/spack-stage/spack-stage-ntpoly-3.1.0-asg2mljsf3vzgz4wisusxbykg5veaku2/spack-build-asg2mlj/CMakeFiles/CMakeScratch/TryCompile-xTeIn1/testCCompiler.c
  >> 34    [spack cc]: Error: SPACK_CC_* variables not set: this usually means a missing `depends_on("c", type="build")` in package.py
  >> 35        gmake[1]: *** [CMakeFiles/cmTC_6390d.dir/build.make:81: CMakeFiles/cmTC_6390d.dir/testCCompiler.c.o] Error 1
     36        gmake[1]: Leaving directory '/local_scratch/sarastel/spack-stage/spack-stage-ntpoly-3.1.0-asg2mljsf3vzgz4wisusxbykg5veaku2/spack-build-asg2mlj/CMakeFiles/CMakeScratch/TryCompile-xTeIn1'
  >> 37        gmake: *** [Makefile:134: cmTC_6390d/fast] Error 2

</pre></details>


### Information on your system

Main tests on the Mahti supercomputer, using the external openblas, openmpi and cmake there. 
Tested also with external GCC 11.4.0, on my local Ubuntu machine, with no other external installs.

```console
$ spack debug report
* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/fee5e77ba2db555f6e7a84ce5a30c44a60b4f094)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/afdf167333306c0a018739a1b27652b5f44d7304
* **Python:** 3.6.8
* **Platform:** linux-rhel8-x86_64_v3
```

&&

```console
$ spack debug report
* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/fee5e77ba2db555f6e7a84ce5a30c44a60b4f094)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/afdf167333306c0a018739a1b27652b5f44d7304
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-icelake
```

### Additional information

CMake fails when configuring ntpoly because the Spack C compiler wrapper cannot run a test compile.

Error message:

```console
[spack cc]: Error: SPACK_CC_* variables not set: this usually means a missing
depends_on("c", type="build") in package.py
```

The ntpoly package currently declares build dependencies for C++ and Fortran but not for C. CMake performs a C compiler test, which fails.

Adding the following line in `repos/spack_repo/builtin/packages/ntpoly/package.py` fixes the issue locally:

    depends_on("c", type="build")

a.ka:

```diff
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
```

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/ntpoly/package.py
+++ b/repos/spack_repo/builtin/packages/ntpoly/package.py
@@ -27,6 +27,7 @@ class Ntpoly(CMakePackage):
 
     variant("shared", default=True, description="Build shared libraries.")
 
+    depends_on("c", type="build")
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
 

